### ISSUE TITLE
Add phantomjs, simplify deployment to production, manage test wihout browsers withing jenkins or docker

### ISSUE THREAD

**Issue by kopax** [mxstbr/react-boilerplate#1299]
I have written a Dockerfile for my application: 

    FROM node:7.2-alpine
  
    RUN mkdir /app
    WORKDIR /app
  
    ADD . /app
    RUN npm install
  
    ENV PATH /app/node_modules/.bin:$PATH
  
    RUN apk update && apk upgrade && \
      apk add --no-cache bash git
  
    RUN apk add --update \
      autoconf
  
    CMD ["npm","run","start:production"]

It's working fine except the probleme is `CHROME_BIN` is undefined and test fail with exit 1.

I guess I will have the same problem if I run the test through jenkins.

What is the correct behavior to have now, I don't wan't to disable the tests in production launch:

I have seen that it was possible to replace chrome by __PhantomJS__ : 

http://stackoverflow.com/questions/34286210/how-to-run-karma-tests-from-docker-container
http://stackoverflow.com/questions/18523947/cannot-install-phantomjs-in-karma

I've try to do it but I am facing this error : 

```
03 12 2016 01:31:02.778:ERROR [preprocess]: Can not load "webpack"!
  TypeError: arguments[i].apply is not a function
    at Compiler.apply (/workspace/github.com/mxstbr/react-...
START:
03 12 2016 01:31:02.871:INFO [karma]: Karma v1.3.0 server started at http://localhost:9876/
03 12 2016 01:31:02.872:INFO [launcher]: Launching browser PanthomJS with unlimited concurrency
03 12 2016 01:31:02.875:ERROR [karma]: Found 1 load error
```

- Is there a way to install chrome on alpine ? I'd like to keep the image as light as possible
- How did you guys solve the problem on jenkins or docker ?

**Comment by mxstbr** [comment]
Have you seen #753? Maybe that helps!

**Comment by tmcdonnell87** [comment]
@kopax I got this working using PhantomJS (using node:6-slim). I basically pulled the files from #753, with minor extensions listed in my comment on that thread

**Comment by easingthemes** [comment]
PhantomJS should be default test browser. 
* Fresh app:
1. My computer will explode while trying to open Chrome
2. PhantomJS needs just few milliseconds

What is main benefit of using Chrome over PhantomJS?

**Comment by gihrig** [comment]
@easingthemes In the latest `dev` branch, soon to be released as 3.4 all that Karma, test in browser stuff has been removed and replaced with the updated [Jest](https://facebook.github.io/jest/). I find that to be a far better dev experience and the project builds (on Travis and AppVeyor) work very well.

It would be interesting to hear feedback on the new test setup from someone deploying to/with Jenkins or Docker.
